server/sonar-server/src/main/java/org/sonar/server/qualitygate/Condition.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualitygate;
+
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class Condition {
+
+  private final String metricKey;
+  private final Operator operator;
+  @CheckForNull
+  private final String warningThreshold;
+  @CheckForNull
+  private final String errorThreshold;
+  private final boolean onLeakPeriod;
+
+  public Condition(String metricKey, Operator operator,
+    @Nullable String errorThreshold, @Nullable String warningThreshold,
+    boolean onLeakPeriod) {
+    this.metricKey = requireNonNull(metricKey, "metricKey can't be null");
+    this.operator = requireNonNull(operator, "operator can't be null");
+    this.onLeakPeriod = onLeakPeriod;
+    this.errorThreshold = errorThreshold;
+    this.warningThreshold = warningThreshold;
+  }
+
+  public String getMetricKey() {
+    return metricKey;
+  }
+
+  public boolean isOnLeakPeriod() {
+    return onLeakPeriod;
+  }
+
+  public Operator getOperator() {
+    return operator;
+  }
+
+  public Optional<String> getWarningThreshold() {
+    return Optional.ofNullable(warningThreshold);
+  }
+
+  public Optional<String> getErrorThreshold() {
+    return Optional.ofNullable(errorThreshold);
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
`Condition` and all `EvaluationQualityGate` related classes implement equals and hashcode to be safely used in tests but mainly because those classes are data objects